tests/common.go

```diff
@@ -4326,6 +4326,38 @@ func CreateBucket(provider string, bucketName string) {
 	})
 }
 
+// IsS3BucketEmpty returns true if bucket empty else false
+func IsS3BucketEmpty(bucketName string) (bool, error) {
+	id, secret, endpoint, s3Region, disableSSLBool := s3utils.GetAWSDetailsFromEnv()
+	sess, err := session.NewSession(&aws.Config{
+		Endpoint:         aws.String(endpoint),
+		Credentials:      credentials.NewStaticCredentials(id, secret, ""),
+		Region:           aws.String(s3Region),
+		DisableSSL:       aws.Bool(disableSSLBool),
+		S3ForcePathStyle: aws.Bool(true),
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to get S3 session to create bucket with %s", err)
+	}
+
+	S3Client := s3.New(sess)
+	input := &s3.ListObjectsInput{
+		Bucket: aws.String(bucketName),
+	}
+
+	result, err := S3Client.ListObjects(input)
+	if err != nil {
+		return false, fmt.Errorf("unable to fetch cotents from s3 failing with %s", err)
+	}
+
+	// Debug log to be removed later
+	log.Info(fmt.Sprintf("Result content %d", len(result.Contents)))
+	if len(result.Contents) > 0 {
+		return false, nil
+	}
+	return true, nil
+}
```

Comment:
How about we return the length of Contents here, and in the test case, we can check if the length of Contents is the same before and after the test case?

Reply:
I don't see any usecase where would want to use the contents. Like these are encrypted files what will you do with them?

Reply:
What if the bucket is not empty initially? To check if DeleteBackup actually worked, cannot we compare the length of the contents?

Reply:
Thanks.